File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dct = dict() # index: target - value

        for i, n in enumerate(nums):
            diff = target - n
            if diff in dct:
                return [dct[diff], i]
            
            dct[n] = i

        return [-1, -1]

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int, int> hs;

        for (const auto elem : nums) {
            if (hs[elem]) {
                return elem;
            }
            else {
                hs[elem]++;
            }
        }
        return -1;
    }
};
